src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/form/main.js

```diff
@@ -60,19 +58,31 @@ define(['app-config', 'config', 'services/sulupreview/preview'], function(AppCon
             this.sandbox.on('sulu.content.navigate', this.navigate, this);
         },
 
+        /**
+         * Initializes the rlp-inputs as well as the rlp-part inputs.
+         * When in edit mode (not in add), the rlp-inputs and the rlp-part inputs
+         * are just like any other inputs and the corresponding promise is resolved right away.
+         */
         initializeResourceLocator: function() {
-            this.startListening = false;
-            this.getDomElementsForTagName('sulu.rlp', function(property) {
-                var element = property.$el.data('element');
-                if (!element || element.getValue() === '' || element.getValue() === undefined || element.getValue() === null) {
-                    this.startListening = true;
-                }
-            }.bind(this));
+            if (!!this.add) {
+                // when a rlp-part gets changed other parts (e.g. submitting) has to wait
+                this.sandbox.dom.on(this.getDomElementsForTagName('sulu.rlp.part'), 'keyup', function() {
+                    this.whenResourceLocatorIsLoaded = $.Deferred();
```

Comment:
This object is generated for every `keyup`? Shouldn't we only do that if the promise haven't been resolved yet?